.github/workflows/php-code-analysis.yml

```diff
@@ -21,42 +31,107 @@ jobs:
                 password: ${{ secrets.github_token }}
 
         steps:
+            -   name: Check if plugin is legacy with target
+                run: |
+                    targets="Core Frontend Backend"
+                    if [ ${{ env.LEGACY_PLUGIN }} = true ]; then
+                        if [ ! -z "${{ env.LEGACY_PLUGIN_TARGET }}" ]; then
+                             for target in $targets; do
+                                if [ $target = ${{ env.LEGACY_PLUGIN_TARGET }} ]
+                                then
+                                    echo "Target set!"
+                                    exit 0
+                                fi
+                            done
+                            echo 'Wrong target has been set! Please set the github action input "legacy-plugin-target" to "Core", "Backend", "Frontend"'
+                            exit 1
+                        fi
+
+                        echo 'No target has been set! Please set the github action input "legacy-plugin-target" to "Core", "Backend", "Frontend"'
+                        exit 1
+                    fi
+                    echo "Not a legacy plugin"
+
             -   run: /usr/bin/supervisord -c /etc/supervisord.conf &
 
-            -   name: Checkout ${{ env.PLUGIN_NAME }} test
+            -   name: Checkout ${{ env.PLUGIN_NAME }}
                 uses: actions/checkout@v3
                 with:
                     path: plugin
+            -   name: Move plugin
+                run: |
+                    if [ ${{ env.LEGACY_PLUGIN }} = true ]; then
+                        mv "$(pwd)/plugin" /shopware/engine/Shopware/Plugins/Local/${{ env.LEGACY_PLUGIN_TARGET }}/${{ env.PLUGIN_NAME }}
+                    else
+                        mv "$(pwd)/plugin" /shopware/custom/plugins/${{ env.PLUGIN_NAME }}
+                    fi
 
-            - name: Get Composer Cache Directory
-              id: composer-cache
-              run: |
-                  echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
-            - uses: actions/cache@v3
-              with:
-                  path: ${{ steps.composer-cache.outputs.dir }}
-                  key: ${{ runner.os }}-composer-${{ hashFiles('plugin/composer.lock') }}
-                  restore-keys: |
-                      ${{ runner.os }}-composer-
+            -   name: Get Composer Cache Directory
```

Comment:
This never worked since the start of the plugin pipeline but went under the radar because it is allowed to fail

I really do not know how to exactly fix this

@PascalThesing can you maybe give an insight here?

i cannot change directory to the plugin directory before executing the composer cache action.

It is nothing all too wild but it would make sense to get this fixed

Reply:
Fixed!